### PR TITLE
[addons][network] add new functions GetHostname(), IsLocalHost(...) and IsHostOnLAN(...) / minor cleanup

### PR DESCRIPTION
**Pull request by AlwinEsch**
## Description

Two new functions where good for inputstream addons.

Commit 1:
------------------------------
**[addons][network] add new functions**

 - GetHostname()
 - IsLocalHost(...)
 - IsHostOnLAN(...)

Commit 2:
------------------------------
**[addons][network] minor cleanups**

Some not needed doc lines are removed, use @ only for doxygen and use
new/delete for his interface structure.

Commit 3:
------------------------------
**[addons][network] separate "C" and "C++" parts of headers**

By them becomes the "C" part placed in own header to see better what language
is where and to make easier for other languages.

Commit 4:
------------------------------
**[addons][network] increase API version to 1.0.1**

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/addons/interfaces/Network.cpp [hunk 1]
```diff
@@ -75,6 +76,39 @@ char* Interface_Network::get_ip_address(void* kodiBase)
...
+    buffer = strdup(hostname.c_str());
+  return buffer;
+}
+
+bool Interface_Network::is_local_host(void* kodiBase, const char* hostname)
+{
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  if (addon == nullptr || hostname == nullptr)
+  {
+    CLog::Log(LOGERROR, "Interface_Network::{} - invalid data (addon='{}', hostname='{}')", __FUNCTION__,
```
Comment (Rechi) [inline xbmc/addons/interfaces/Network.cpp:104]: > The `ColumnLimit` in `.clang-format` is set to `100` which defines line length [...]

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#3-formatting

**Review by Rechi — COMMENTED** [review]
Please follow the [current code guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md).

**Review by phunkyfish — COMMENTED** [review]
Looks good apart from the one clang format issue Rechi pointed out.

**Comment by AlwinEsch** [comment]
Have it updated and added also `IsHostOnLAN(...)` to there.

Further splitted like with PVR now the "C" part to own header.

**Comment by AlwinEsch** [comment]
jenkins build this please

**Comment by AlwinEsch** [comment]
Runtime with this, was OK:
```cpp
bool RuntimeTest_Network()
{
  bool ret;

  std::string ip = kodi::network::GetIPAddress();
  fprintf(stderr, "- kodi::network::GetIPAddress():               %s\n", ip.c_str());

  std::string hostname = kodi::network::GetHostname();
  fprintf(stderr, "- kodi::network::GetHostname():                %s\n", hostname.c_str());

  ret = kodi::network::IsLocalHost("127.0.0.1");
  fprintf(stderr, "- kodi::network::IsLocalHost(\"127.0.0.1\"):   %s\n", ret ? "yes" : "no");
  ret = kodi::network::IsLocalHost("localhost");
  fprintf(stderr, "- kodi::network::IsLocalHost(\"localhost\"):   %s\n", ret ? "yes" : "no");
  ret = kodi::network::IsLocalHost("192.168.0.1");
  fprintf(stderr, "- kodi::network::IsLocalHost(\"192.168.0.1\"): %s\n", ret ? "yes" : "no");
  ret = kodi::network::IsLocalHost("212.4.0.1");
  fprintf(stderr, "- kodi::network::IsLocalHost(\"212.4.0.1\"):   %s\n", ret ? "yes" : "no");

  std::string encodedUrl = kodi::network::URLEncode("François");
  fprintf(stderr, "kodi::network::URLEncode(\"François\")        '%s'\n", encodedUrl.c_str());

  std::string ipAddress;
  ret = kodi::network::DNSLookup("www.google.com", ipAddress);
  fprintf(stderr, "DNSLookup returned for www.google.com the IP '%s', call was %s\n", ipAddress.c_str(), ret ? "ok" : "failed");
  return true;
}
```

**Pull request merged**